scripts/report_scripts/index.Rmd

```diff
@@ -489,178 +531,250 @@ if (nrow(mutations_sig_unfiltered) > 0) {
 }
 ```
 
-```{r text_mutations_pooled, results='asis', eval=APPROVED_MUTATIONS_FOUND}
+```{r filter_pool_mutations, eval=RUN_MUTATION_REGRESSION}
+# filter good samples for only mutations with sig pvalues for plotting
+filtered_approved_mut_plot <- dplyr::select(
+  approved_mut_plot,
+  c(
+    samplename,
+    dates,
+    location_name,
+    coordinates_lat,
+    coordinates_long,
+    mutations_sig$mutation
+  )
+)
 
-cat('## Trending mutations over time {.tabset .tabset-fade}
+# check if the filtered df has actual values besides meta data
+APPROVED_MUTATIONS_FOUND <- (length(filtered_approved_mut_plot) > 5) &&
+  (nrow(filtered_approved_mut_plot) > 0)
 
-These plots show the relative frequency of detected mutations with strong increasing trends in samples at specific wastewater sampling locations and how the frequencies change over time. Please use the tabs to access the not-pooled plots for each location.')
+approved_mut_plot_location_pooled <- pool_by_mean(filtered_approved_mut_plot,
+  na_handling = TRUE,
+  group_fun = "day_location"
+)
 ```
 
+```{r mutation_trends_pooled_text, results='asis', eval=APPROVED_MUTATIONS_FOUND}
+cat(
+  "## Trending mutations over time {.tabset .tabset-fade}\n\n",
+  "These plots show the relative frequency of detected mutations with strong ",
+  "increasing trends in samples at specific wastewater sampling locations and ",
+  "how the frequencies change over time. Please use the tabs to access the ",
+  "not-pooled plots for each location.",
+  sep = ""
+)
+```
 
 
-
-
-```{r mutations_pooled, eval=APPROVED_MUTATIONS_FOUND}
-
+```{r mutation_trends_pooled_plot_gen, eval=APPROVED_MUTATIONS_FOUND}
 melted <- reshape2::melt(approved_mut_plot_location_pooled,
-                         id.vars = c("dates", "samplename"))
+  id.vars = c("dates", "samplename")
+)
+
 melted <- melted %>%
-# flag if mutation is signature mutation
+  # flag if mutation is signature mutation
   rowwise() %>%
-  mutate(sigflag = ifelse(any(sigmuts.df == str_split(variable, "::")[[1]][2]), "SigMut", '')) %>%
+  mutate(sigflag = ifelse(any(
+    sigmuts.df == str_split(variable, "::")[[1]][2]
```

Comment:
I know this is just a formatting change, but I can't help myself: splitting strings at arbitrary separators and picking numbered fields from the split result .... that's just ripe for errors.

Reply:
This will be changed to `str_detect` in a future PR.